core/src/main/java/com/spotify/ffwd/output/CoreOutputManager.java

```diff
@@ -176,10 +234,25 @@ public void sendBatch(Batch batch) {
 
         int batchSize = batch.getPoints().size();
 
-        if (batchSize > 0 && !rateLimitAllowed(batchSize, batch.getPoints().get(0).getKey())) {
-            log.debug("Dropping {} metrics due to rate limiting", batchSize);
-            statistics.reportMetricsDroppedByRateLimit(batchSize);
-            return;
+        hyperLog.offer(batch.hashCode());
+        statistics.reportMetricsCardinality(batch.hashCode());
+
+        if (!Arrays.asList(KEYS_NEVER_TO_DROP).contains(batch.getPoints().get(0).getKey())) {
```

Comment:
this code for `batch` is nearly identical to the one for a single metric. can the code be deduplicated?

Reply:
i've consolidated it